5 homepage frontend

AntonGrig opened the pull request.

DenisYolo committed:
signin and index page added.

DenisYolo committed:
test

Commit:
Adding index and signin html

Commit:
Adding index and signin html

Commit:
Created initial signup endpoint.
Further additions might be needed.

PackageL committed:
Added:
event-list.html
user-events.html
view-event.html

Made adjustments in:
create-event.html - added base code
Event.java - change data type to LocalData
event-list.html - added some base code
EventController.java - added mappings
index.html - made few adjustments
signin.html - made few adjustments
UserController.java - removed Event mappings

PackageL committed:
Added:
event-list.html
user-events.html
view-event.html

Made adjustments in:
create-event.html - added base code
Event.java - change data type to LocalData
event-list.html - added some base code
EventController.java - added mappings
index.html - made few adjustments
signin.html - made few adjustments
UserController.java - removed Event mappings. Added submit mapping
signup.html - restructured the code and added styling

PackageL commented:
event-list.html
user-events.html
view-event.html

Made adjustments in:
create-event.html - added base code
Event.java - change data type to LocalData
event-list.html - added some base code
EventController.java - added mappings
index.html - made few adjustments
signin.html - made few adjustments
UserController.java - removed Event mappings. Added submit mapping
signup.html - restructured the code and added styling

Commit:
Made next changes:
- Created EventService interfaces and their implementation
- EventRepository and UserRepository should be called in service layer, EventService and UserService should be called in controller layer. Made necessary changes.
- Post mapping was not working, because html file still had old endpoint written in the action.
- Added Role enum file. Implementing on the way.
- Added Lombok to our pom.xml file. Watch how it is used in User.java

Luke586 committed:
Merge branch 'main' into 5-homepage-frontend
